www/include/reporting/dashboard/csvExport/csv_HostLogs.php

```diff
@@ -62,111 +60,175 @@
+    . $hostStats["MAINTENANCE_T"] . "s;"
+    . $hostStats["MAINTENANCE_TP"] . "%;;;\n";
+
+echo _("UNDETERMINED") . ";"
+    . $hostStats["UNDETERMINED_T"] . "s;"
+    . $hostStats["UNDETERMINED_TP"] . "%;\n";
 echo "\n";
 echo "\n";
 
-echo _("Service") . ";" . _("OK") . " %; " . _("OK") ." Alert;"
+echo _("Service") . ";"
+    . _("OK") . " %; " . _("OK") ." Alert;"
     . _("Warning") . " %;" . _("Warning") . " Alert;"
     . _("Critical") . " %;" . _("Critical") . " Alert;"
     . _("Unknown") . " %;" . _("Unknown") . " Alert;"
     . _("Scheduled Downtimes") . " %;" . _("Undetermined") . "%;\n";
 
-$hostServicesStats =  getLogInDbForHostSVC($id, $start_date, $end_date, $reportingTimePeriod);
+$hostServicesStats =  getLogInDbForHostSVC(
+    $hostId,
+    $start_date,
+    $end_date,
+    $reportingTimePeriod
+);
+
 foreach ($hostServicesStats as $tab) {
     if (isset($tab["DESCRIPTION"]) && $tab["DESCRIPTION"] != "") {
-        echo $tab["DESCRIPTION"] . ";" . $tab["OK_TP"] . " %;" . $tab["OK_A"] . ";"
-            . $tab["WARNING_TP"] . " %;" . $tab["WARNING_A"] . ";"
-            . $tab["CRITICAL_TP"] . " %;" . $tab["CRITICAL_A"] . ";"
-            . $tab["UNKNOWN_TP"] .  "%;" . $tab["UNKNOWN_A"] . ";"
-            . $tab["MAINTENANCE_TP"] . " %;" . $tab["UNDETERMINED_TP"] . "%;\n";
+        echo $tab["DESCRIPTION"] . ";"
+            . $tab["OK_TP"] . " %;"
+            . $tab["OK_A"] . ";"
+            . $tab["WARNING_TP"] . " %;"
+            . $tab["WARNING_A"] . ";"
+            . $tab["CRITICAL_TP"] . " %;"
+            . $tab["CRITICAL_A"] . ";"
+            . $tab["UNKNOWN_TP"] .  "%;"
+            . $tab["UNKNOWN_A"] . ";"
+            . $tab["MAINTENANCE_TP"] . " %;"
+            . $tab["UNDETERMINED_TP"] . "%;\n";
     }
 }
-
 echo "\n";
 echo "\n";
 
-/*
- * Evolution of host availability in time
- */
-echo _("Day") . ";" . _("Duration").";"
-    . _("Up") . " (s);" . _("Up")." %;" . _("Up") ." "._("Alert") . ";"
-    . _("Down") . " (s);" . _("Down") . " %;" . _("Down") . " " . _("Alert") . ";"
-    . _("Unreachable") . " (s);" . _("Unreachable") . " %;" . _("Unreachable") . " " . _("Alert")
+// Evolution of host availability in time
+echo _("Day") . ";"
+    . _("Duration") . ";"
+    . _("Up") . " (s);"
+    . _("Up") . " %;"
+    . _("Up") ." " . _("Alert") . ";"
+    . _("Down") . " (s);"
+    . _("Down") . " %;"
+    . _("Down") . " " . _("Alert") . ";"
+    . _("Unreachable") . " (s);"
+    . _("Unreachable") . " %;"
+    . _("Unreachable") . " " . _("Alert")
     . _("Day") . ";\n";
-$rq = "SELECT  * " .
-        "FROM `log_archive_host` " .
-        "WHERE `host_id` = '".$id."' " .
-        "AND `date_start` >= '".$start_date."' " .
-        "AND `date_end` <= '".$end_date."' " .
-        "ORDER BY `date_start` desc";
-$DBRESULT = $pearDBO->query($rq);
-
-while ($row = $DBRESULT->fetchRow()) {
+
+$dbResult = $pearDBO->query(
+    "SELECT  * FROM `log_archive_host` " .
+    "WHERE `host_id` = '" . $hostId . "' " .
```

Comment:
In this case, using a prepare statement is hightly recommended

Reply:
Refacto the whole code is out of my scope... ;)